### ISSUE TITLE
Add ARIA roles to HTML

### ISSUE THREAD

**Issue by nlhkabu** [pypa/warehouse#810]
From our accessibility audit by @xavierdutreilh.

We should add [ARIA](http://www.w3.org/TR/wai-aria/roles#landmark_roles) roles to our HTML to make it more accessible for users using screen readers.

Of particular concern are:
- The search form (`<form ... role="search">`)
- The vertical tabs on the packages page (http://www.w3.org/TR/wai-aria/roles#tab)

We also need to add `aria-hidden="true"` to the link icons on the package page as they are not readable by screen readers.  See https://github.com/FortAwesome/Font-Awesome/issues/6133 for more information.

**Comment by souravsingh** [comment]
I would like to work on this. How do I start?

**Comment by nlhkabu** [comment]
Hi @souravsingh.  I'd suggest that you read about the ARIA specifications to understand what they are and why we would use them.  Then it is just a matter of actually adding them to the HTML.

I would do it in this order:

1.  Add `aria-hidden="true"` to icons (on the package detail page)
2.  Add the `search` role to each search form
3.  Add `tab` role to the vertical tabs on the packages page

**Comment by xavierdutreilh** [comment]
@souravsingh, do you have a screen reader installed on your computer? If you do have a recent operating system, chances are that you already got one. Then, you can clone the repository and set up the project on your machine. After that, open the development version in your web browser and run the screen reader.

You will see the screen reader trying to read icons and fail. For instance, VoiceOver on OSX reads: `▢ Latest Version` on the package detail page, instead of `Latest Version` only.

If you need an easier ticket to start with, take a look at #861.

**Comment by nlhkabu** [comment]
Removing the 'easy' label as @xavierdutreilh has told me that the vertical tab section of this issue is a little complex.

**Comment by xavierdutreilh** [comment]
As discussed with @nlhkabu, the vertical tab section is not an easy one. So, I'll deal with it in the next few days

**Comment by dstufft** [comment]
I'm gonna drop this out of the Launch milestone, not because I don't think it's important, but because legacy PyPI doesn't currently have this, so I think it can be added at any time before or after the launch.
